haystack/pipelines/base.py

```diff
@@ -390,33 +394,63 @@ def eval(
             for node_name in predictions["_debug"].keys():
                 node_output = predictions["_debug"][node_name]["output"]
                 df = self._build_eval_dataframe(
-                    query, label, node_name, node_output)
+                    query, label, node_name, node_output, sas_model_name_or_path)
                 eval_result.append(node_name, df)
 
         return eval_result
 
-    def _build_eval_dataframe(self, query: str, labels: MultiLabel, node_name: str, node_output: dict) -> DataFrame:
-        answer_cols = ["answer", "document_id", "offsets_in_document"]
+    def _build_eval_dataframe(self, 
```

Comment:
Please add a few more comments in the code that help others to navigate quickly later on. We saw that in the old eval code that it can become very tricky after a few months when coming back for maintenance or smaller fixes.

Reply:
done